core/src/main/scala/fs2/gzip/package.scala

```diff
@@ -126,28 +127,19 @@ package object gzip {
               }
 
               if (len > 0)
-                Stream.chunk(Chunk.bytes(inner, 0, len)).covary[F]
-              else
-                Stream.empty[F]
-            }
-          }
-
-          // drains abis through gzis
-          @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
-          lazy val repeatedlyDecompress: Stream[F, Byte] = Stream force {
-            Sync[F] delay {
-              if (abis.available() > 0)
-                stepDecompress ++ repeatedlyDecompress
+                Stream.chunk(Chunk.bytes(inner, 0, len)).covary[F] ++ stepDecompress
               else
                 Stream.empty[F]
             }
           }
 
+          // TODO: It is possible for this to fail with a non-progressive error
+          //       if `in` contains bytes in addition to the compressed data.
           val mainline = in.chunks flatMap { chunk =>
             Stream.eval_(push(chunk)) ++ stepDecompress
```

Comment:
Shouldn't we remove the `++ stepDecompress` here?

Reply:
We need it to actually emit the decompressed bytes. The stream looks like `push(chunk1) >> <decompressed bytes from chunk1> >> push(chunk2) >> <decompressed bytes from chunk2> ...`

Reply:
Ooooh, because this is on the flatmap. Duh.